docs/Guide/application-commands/interaction-handlers/autocomplete.mdx

```diff
@@ -3,6 +3,84 @@ sidebar_position: 2
 title: Autocomplete
 ---
 
-<!-- INFO ON AUTOCOMPLETE INTERACTION HANDLERS AS WELL AS HOW TO USE AUTOCOMPLETERUN -->
+Certain Chat Input Command options can be autocompleted based on what the user has written. There are two ways you can
+create an autocomplete interaction handler. The first way is by creating `Piece` that extends
+[`InteractionHandler`][interaction-handler] and the second way is by implementing the
+[`autocompleteRun`][autocomplete-run] method.
 
-<!-- TODO: https://vladfrangu.notion.site/Application-Commands-3d1683f23f0a4fa6b3e682bc8c1cc1b9#ceb1a6f5ec154f85987a95181d512660 -->
+:::warning
+
+You cannot defer auto-completes (that would be silly)! You **must** respond to the interaction within **3 seconds**.
+
+:::
+
+## Interaction Handler
+
+This is no different than the rest - extend the [`InteractionHandler`][interaction-handlers] method and give it a
+`parse` and `run` method. Note here that you have access to `<AutocompleteInteraction>.commandName` or
+`<AutocompleteInteraction>.commandId` to differentiate the command the user is using.
+
+Example:
+
+```typescript ts2esm2cjs
+import { InteractionHandler, InteractionHandlerTypes, PieceContext } from '@sapphire/framework';
+import type { AutocompleteInteraction } from 'discord.js';
+
+export class AutocompleteHandler extends InteractionHandler {
+  public constructor(ctx: PieceContext, options: InteractionHandler.Options) {
+    super(ctx, {
+      ...options,
+      interactionHandlerType: InteractionHandlerTypes.Autocomplete
+    });
+  }
+
+  public override async run(interaction: AutocompleteInteraction, result: InteractionHandler.ParseResult<this>) {
+    return interaction.respond(result);
+  }
+
+  public override async parse(interaction: AutocompleteInteraction) {
+    // Only run this interaction for the command with ID '1000802763292020737'
+    if (interaction.commandId !== '1000802763292020737') return this.none();
+
+    // Get the focussed (current) option
+    const focusedOption = interaction.options.getFocused(true);
+
+    // Ensure that the option name is one that can be autocompleted, or return none if not.
+    switch (focusedOption.name) {
+      case 'search': {
+        // Search your API or similar. This is example code!
+        const searchResult = await myApi.searchForSomething(focusedOption.value as string);
+
+        // Map the search results to the structure required for Autocomplete
+        return this.some(searchResult.map((match) => ({ name: match.name, value: match.key })));
+      }
+      default:
+        return this.none();
+    }
```

Comment:
I'd move this to the run method... Or alternatively uhhhh..I dunno, it feels wrong to have it in parse

Reply:
You wah? You used to campaign for having this in parse *hard* at first when you first wrote all of that. You even talk about it on your own notion doc:
![image](https://user-images.githubusercontent.com/4019718/183155831-9a442ec8-dcfc-49b0-b2eb-a613e68f3a89.png)

Reply:
Yeah lets keep it as is, it just felt weird cuz the run method was so bland now XD